tests/baselines/reference/unconstrainedTypeParameterNarrowing.js

```diff
@@ -0,0 +1,55 @@
+//// [unconstrainedTypeParameterNarrowing.ts]
+function f1<T>(x: T) {
+    if (typeof x === "object" && x) {
+        g(x);
+    }
+}
+
+function f2<T extends unknown>(x: T) {
+    if (typeof x === "object" && x) {
+        g(x);
+    }
+}
+
+// #48468 but with an explicit constraint so as to not trigger the `{}` and unconstrained type parameter bug
+function deepEquals<T extends unknown>(a: T, b: T) {
```

Comment:
Can you have a test for both the explicit and non-explicit constraint?

FWIW I have tests at #48576.